x-pack/legacy/plugins/lens/public/indexpattern_plugin/indexpattern_suggestions.tsx

```diff
@@ -298,17 +298,73 @@ function createNewLayerWithMetricAggregation(
 export function getDatasourceSuggestionsFromCurrentState(
   state: IndexPatternPrivateState
 ): Array<DatasourceSuggestion<IndexPatternPrivateState>> {
-  const layers = Object.entries(state.layers);
+  const layers = Object.entries(state.layers || {});
 
-  return layers
-    .map(([layerId, layer], index) => {
+  return _.flatten(
```

Comment:
Nit: You're kind of doing a map and filter inside a single map with your if statement on line 305. I wonder if it would be easier to follow if we did a filter?

```js
return _(Object.entries(state.layers || {}))
  .filter(([_id, layer]) => layer.columnOrder.length)
  .map(([layerId, layer]) => {
    const onlyMetric = layer.columnOrder.every(columnId => !layer.columns[columnId].isBucketed);
    const onlyBucket = layer.columnOrder.every(columnId => layer.columns[columnId].isBucketed);
    if (onlyMetric || onlyBucket) {
      // intermediary chart, don't try to suggest reduced versions
      return buildSuggestion({ state, layerId, isMultiRow: false, datasourceSuggestionId: index });
    }

    return createSimplifiedTableSuggestions(state, layerId);
  })
  .flatten()
  .map(assignDatasourceSuggestionIds)
  .value();
```